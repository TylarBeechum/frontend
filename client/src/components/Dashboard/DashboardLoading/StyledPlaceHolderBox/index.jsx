import React from 'react';
import styled from 'styled-components';
import { Box } from 'grommet';

const index = ({ ...rest }) => {
  return (
    <AnimationBox
      animation={{
        type: 'fadeOut',
        duration: 1750
      }}
      background="light-4"
      round
      {...rest}
    />
  );
};

const AnimationBox = styled(Box)`
  animation-iteration-count: infinite;
`;

export default index;
