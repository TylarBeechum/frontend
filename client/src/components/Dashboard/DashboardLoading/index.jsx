import React from 'react';
import styled from 'styled-components';
import { Grid } from 'grommet';

import AnimatedBox from './StyledPlaceHolderBox';

const index = () => {
  return (
    <DashboardLoadingContainer>
      <Grid
        areas={[
          { name: 'card0', start: [0, 0], end: [2, 1] },
          { name: 'card1', start: [0, 1], end: [2, 1] },
          { name: 'card2', start: [0, 2], end: [2, 2] },
          { name: 'card3', start: [0, 3], end: [2, 3] },
          { name: 'card4', start: [0, 4], end: [2, 4] }
        ]}
        columns={['small', 'flex', 'small']}
        rows={['110px', '110px', '110px', '110px', '110px']}
        gap="small"
      >
        <AnimatedBox gridArea="card0" background="light-2" />
        <AnimatedBox gridArea="card1" background="light-2" />
        <AnimatedBox gridArea="card2" background="light-2" />
        <AnimatedBox gridArea="card3" background="light-2" />
        <AnimatedBox gridArea="card4" background="light-2" />
      </Grid>
    </DashboardLoadingContainer>
  );
};

const DashboardLoadingContainer = styled.div`
  margin: 20px auto 0;
  max-width: 1600px;
  width: 100%;
`;

export default index;
