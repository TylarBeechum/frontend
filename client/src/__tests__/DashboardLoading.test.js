/* eslint-disable */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import {
  renderWithRouterAndProviders,
  rendererWithRouterAndProviders
} from '../__testHelpers__';

import DashboardLoading from '../components/Dashboard/DashboardLoading';

describe('<DashboardLoading />', () => {
  describe('should render the component successfully without crashing', () => {
    it('tests that the component does not crash when it is rendered', () => {
      const helpers = renderWithRouterAndProviders(<DashboardLoading />);
    });
  });

  describe('should match the previous snapshot of the component', () => {
    it('tests that the component has not changed and matched the previous snapshot', () => {
      const tree = rendererWithRouterAndProviders(<DashboardLoading />);
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
