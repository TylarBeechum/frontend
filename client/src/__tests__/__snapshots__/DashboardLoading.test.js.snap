// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardLoading /> should match the previous snapshot of the component tests that the component has not changed and matched the previous snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 iDwiuh"
>
  <div
    className="sc-bZQynM foLbtC"
  >
    <div
      className="StyledGrid-sc-1wofa1l-0 gFgmqz"
    >
      <div
        className="sc-EHOje fCtvTT StyledBox-sc-13pk1d4-0 bjbrBv"
      />
      <div
        className="sc-EHOje fCtvTT StyledBox-sc-13pk1d4-0 eVxtVF"
      />
      <div
        className="sc-EHOje fCtvTT StyledBox-sc-13pk1d4-0 uOYyC"
      />
      <div
        className="sc-EHOje fCtvTT StyledBox-sc-13pk1d4-0 gvfGEG"
      />
      <div
        className="sc-EHOje fCtvTT StyledBox-sc-13pk1d4-0 kNCNnG"
      />
    </div>
  </div>
</div>
`;
